### PR TITLE
Major logging overhaul

### PR DESCRIPTION
**Pull request by rbren**
**End-user friendly description of the problem this fixes or functionality that this introduces**

I'm declaring log bankruptcy, and changing all info logs to debug logs

bankruptcy scene from the office

- [x] Include this change in the Release Notes. If checked, you must provide an **end-user friendly** description for your change below
Pared down the backend logging to make it less verbose and more informative

---
**Give a summary of what the PR does, explaining any non-trivial design decisions**

Our logs have gotten a little wild.

This changes _every_ info log into a debug log. Maybe too extreme, but I'm happy to revert specific instances that people care about

Here's what the INFO logs look like now:
```bash
❯ poetry run uvicorn openhands.server.listen:app --host 0.0.0.0 --port 3000 --reload
INFO:     Will watch for changes in these directories: ['/home/rbren/git/openhands']
INFO:     Uvicorn running on http://0.0.0.0:3000 (Press CTRL+C to quit)
INFO:     Started reloader process [2545852] using WatchFiles
INFO:     Started server process [2545856]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     ('127.0.0.1', 34392) - "WebSocket /ws" [accepted]
2024-10-25 14:41:15,629 - INFO - New session: f33a4b35-651c-46c8-be7b-49130c6bc106
INFO:     connection open
2024-10-25 14:41:15,754 - INFO - Building image: ghcr.io/all-hands-ai/runtime:oh_v0.11.0_iwllm0juriwk9jfp_slzg6paebd885wfv
2024-10-25 14:41:15,757 - INFO - [runtime f33a4b35-651c-46c8-be7b-49130c6bc106] Starting runtime with image: ghcr.io/all-hands-ai/runtime:oh_v0.11.0_iwllm0juriwk9jfp_slzg6paebd885wfv
2024-10-25 14:41:16,126 - INFO - [runtime f33a4b35-651c-46c8-be7b-49130c6bc106] Container started: openhands-runtime-f33a4b35-651c-46c8-be7b-49130c6bc106
2024-10-25 14:41:16,127 - INFO - [runtime f33a4b35-651c-46c8-be7b-49130c6bc106] Waiting for client to become ready at http://host.docker.internal:35416...
2024-10-25 14:41:30,203 - INFO - [runtime f33a4b35-651c-46c8-be7b-49130c6bc106] Runtime is ready.
2024-10-25 14:41:30,214 - ERROR - Failed to restore state from session: [Errno 2] No such file or directory: '/tmp/file_store/sessions/f33a4b35-651c-46c8-be7b-49130c6bc106/agent_state.pkl'
2024-10-25 14:41:30,215 - INFO - [Agent Controller f33a4b35-651c-46c8-be7b-49130c6bc106] Starting step loop...
2024-10-25 14:41:30,216 - INFO - [Agent Controller f33a4b35-651c-46c8-be7b-49130c6bc106] Setting agent(CodeActAgent) state from AgentState.LOADING to AgentState.INIT
2024-10-25 14:41:30,224 - INFO - [Agent Controller f33a4b35-651c-46c8-be7b-49130c6bc106] Setting agent(CodeActAgent) state from AgentState.INIT to AgentState.RUNNING
INFO:     127.0.0.1:60188 - "GET /api/list-files HTTP/1.1" 200 OK
INFO:     127.0.0.1:60204 - "GET /api/list-files HTTP/1.1" 200 OK
INFO:     127.0.0.1:60214 - "GET /api/list-files HTTP/1.1" 200 OK
2024-10-25 14:41:31,320 - INFO - [Agent Controller f33a4b35-651c-46c8-be7b-49130c6bc106] LEVEL 0 LOCAL STEP 0 GLOBAL STEP 0
2024-10-25 14:41:37,281 - INFO - [Agent Controller f33a4b35-651c-46c8-be7b-49130c6bc106] LEVEL 0 LOCAL STEP 1 GLOBAL STEP 1
2024-10-25 14:41:40,523 - INFO - [Agent Controller f33a4b35-651c-46c8-be7b-49130c6bc106] Setting agent(CodeActAgent) state from AgentState.RUNNING to AgentState.FINISHED
INFO:     127.0.0.1:54222 - "GET /api/list-files HTTP/1.1" 200 OK

```

---
**Link of any specific issues this addresses**

### CODE REVIEW

File: openhands/runtime/action_execution_server.py [hunk 1]
```diff
@@ -586,7 +586,5 @@ async def list_files(request: Request):
             logger.error(f'Error listing files: {e}', exc_info=True)
             return []
 
-    logger.info('Runtime client initialized.')
-
-    logger.info(f'Starting action execution API on port {args.port}')
+    logger.debug(f'Storting action execution API on port {args.port}')
```
Comment (rbren) [inline openhands/runtime/action_execution_server.py:589]:
```suggestion
    logger.debug('Starting action execution API on port {args.port}')
```

File: openhands/runtime/base.py [hunk 2]
```diff
@@ -104,7 +108,7 @@ def add_env_vars(self, env_vars: dict[str, str]) -> None:
                 code += f'os.environ["{key}"] = {json.dumps(value)}\n'
             code += '\n'
             obs = self.run_ipython(IPythonRunCellAction(code))
-            logger.info(f'Added env vars to IPython: code={code}, obs={obs}')
+            logger.debug(f'Added env vars to IPython: code={code}, obs={obs}')
```
Comment (rbren) [inline openhands/runtime/base.py:111]:
```suggestion
            self.log('debug', f'Added env vars to IPython: code={code}, obs={obs}')
```

File: openhands/server/listen.py [hunk 3]
```diff
@@ -288,7 +288,7 @@ async def websocket_endpoint(websocket: WebSocket):
         ```
     - Run an IPython command:
         ```json
-        {"action": "run_ipython", "args": {"command": "print('Hello, IPython!')"}}
+        {"action": "run_ipython", "args": {"command": "logger.debug('Hello, IPython!')"}}
```
Comment (rbren) [inline openhands/server/listen.py:291]:
```suggestion
        {"action": "run_ipython", "args": {"command": "print('Hello, IPython!')"}}
```

File: openhands/utils/chunk_localizer.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def create_chunks(
     try:
         parser = get_parser(language) if language is not None else None
     except AttributeError:
-        # print(f"Language {language} not supported. Falling back to raw string.")
+        # logger.debug(f"Language {language} not supported. Falling back to raw string.")
```
Comment (rbren) [inline openhands/utils/chunk_localizer.py:46]:
```suggestion
        logger.debug(f"Language {language} not supported. Falling back to raw string.")
```

**Comment by enyst** [comment]
This one is a little strange actually

> ERROR - Failed to restore state

I think the info is useful, but I keep seeing it worry users more than it should. At the very least, it could say 'warning', or even less than a warning.

**Review by tofarr — COMMENTED** [review]
Nice to see this one! Good to get those print statements gone too. I see also the rejig of the init method in the remote runtime. 🍰 

One question for now: Are we sure we want to add logging to the contract for the agent controller rather than just having an associated logger?

One question for future consideration: We use the same logger for everything. Would it make sense to split these up?
One pattern that worked quite well for me in the past was to have a logger per package. This allowed setting the global level to debug, and then turning on logging for particular parts of the system for short periods of time for debugging. It was useful when turning on info would have resulted in getting ~1000 log messages a second.

**Review by enyst — APPROVED** [review]

**Comment by tobitege** [comment]
@rbren  In `eventstream_runtime.py` the use of `log_buffer.client_ready` has been completely removed.
What the code now effectively only checks is if the _container_ is ready, but not if the _client_ is ready, which meant to be _after_ plugin installations, user creation and bash env being set etc.
The latter should be done _before_ any commands/actions would be sent to the container for execution.

**Comment by rbren** [comment]
@tobitege I'm not sure what changed here, but by the time the `/alive` endpoint is working the client is totally ready! So the check there was duplicative

In more detail, we were waiting for this log: https://github.com/All-Hands-AI/OpenHands/blob/6d19c93d192bc9d816672814993c206825dbceeb/openhands/runtime/action_execution_server.py#L590

But the `/alive` endpoint won't return until _after_ that log, so `/alive` captures the `client_ready` status as well

**Pull request merged**